quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractUpdatesCommand.java

```diff
@@ -97,4 +101,31 @@ void printFeatureDisabled() {
         printError("Unable to use this command. The preview feature 'rolling-updates' is not enabled.");
     }
 
+    static Map<String, CompatibilityMetadataProvider> loadAllProviders() {
+        Map<String, CompatibilityMetadataProvider> providers = new HashMap<>();
+        for (var p : ServiceLoader.load(CompatibilityMetadataProvider.class)) {
+            providers.merge(p.getId(), p, (existing, current) -> {
+                if (existing.priority() == current.priority()) {
+                    throw new IllegalArgumentException("Unable to handle two providers with the same id (%s) and priority.".formatted(existing.getId()));
+                }
+                // If a user wants to replace default providers with their own.
+                return existing.priority() < current.priority() ?
+                        current :
+                        existing;
+            });
+        }
+        return providers;
+    }
+
+    private static void loadConfiguration() {
+        // Initialize config
+        var configProvider = ServiceLoader.load(ConfigProviderFactory.class)
```

Comment:
@mhajas, I'm unsure if I should use the `ServiceLoader`. I'm tempted to use `Config.init(new MicroProfileConfigProvider());` as it is done in the `loadFactories()` methods below

https://github.com/keycloak/keycloak/blob/a04d3ec57f405e6a11e84423ecdbe70344584efc/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java#L689

Reply:
I like your suggestion. I was following what we do here: https://github.com/keycloak/keycloak/blob/b2925f36e4a658f61719759952f49ede3a95de77/services/src/main/java/org/keycloak/services/resources/KeycloakApplication.java#L180-L182 but I think it makes more sense to follow `KeycloakProcessor` here. So let's go with your suggestion. Maybe we should get opinion from CND team on this.

Reply:
I saw that method, too. It is only used by `ResteasyKeycloakApplication` and this feature doesn't make any usage of the web interface. I updated the code to use `Config.init(new MicroProfileConfigProvider()); `.